src/jigsaw/common/directive/ribbon/ribbon.ts

```diff
@@ -16,89 +16,96 @@ type Position = BasePosition & { ribbon?: Style }
 @Directive({
     selector: '[jigsawRibbon], [jigsaw-ribbon]'
 })
-export class JigsawRibbonDirective extends JigsawBadgeBase {
-    private _removeRibbonClickHandler: Function;
+export class JigsawRibbonDirective extends BadgeBsae {
 
     constructor(protected _elementRef: ElementRef, protected _render: Renderer2, protected _zone?: NgZone,) {
         super(_elementRef, _render, _zone);
     }
 
+    private _removeRibbonClickHandler: Function;
+
     /**
      勋带内容
      */
-    private _jigsawRibbonValue: string;
-    @Input()
-    public get jigsawRibbonValue(): string {
-        return this._jigsawRibbonValue
+
+    public value: string;
+
+    @Input('jigsawRibbonValue')
+    get jigsawRibbonValue(): string {
+        return this.value
     }
 
-    public set jigsawRibbonValue(value: string) {
-        if (this._jigsawRibbonValue != value) {
-            this._jigsawRibbonValue = value;
+    set jigsawRibbonValue(_value: string) {
+        if (this.value != _value) {
+            this.value = _value;
             this._addRibbon();
         }
     }
 
     /**
      *  勋带大小
      * */
-    private _jigsawRibbonSize: 'large' | 'normal' | 'small' = 'normal';
-    @Input()
-    public get jigsawRibbonSize() {
-        return this._jigsawRibbonSize;
-    }
+    public size: 'large' | 'normal' | 'small' = 'normal';
 
-    public set jigsawRibbonSize(size) {
-        if (this._jigsawRibbonSize != size) {
-            this._jigsawRibbonSize = size
+    @Input('jigsawRibbonSize')
+    get jigsawRibbonSize() {
+        return this.size;
+    }
+    set jigsawRibbonSize(_size) {
+        if (this.size != _size) {
+            this.size = _size
             this._addRibbon();
         }
     }

```

Comment:
private _size: 'large' | 'normal' | 'small' = 'normal';

    get size(): "large" | "normal" | "small" {
        return this._size;
    }

    set size(value: "large" | "normal" | "small") {
        if (this.size != value) {
            this.size = value
            this._addRibbon();
        }
    }